rapids_build_backend/impls.py

```diff
@@ -118,37 +118,25 @@ def _get_git_commit() -> Union[str, None]:
 
 
 @contextmanager
-def _edit_git_commit(config):
+def _write_git_commit(config, project_name):
     """
     Temporarily modify the git commit of the package being built.
 
     This is useful for projects that want to embed the current git commit in the package
     at build time.
     """
     commit_file = config.commit_file
+    if commit_file == "":
+        commit_file = os.path.join(project_name.replace("-", "_"), "GIT_COMMIT")
     commit = _get_git_commit()
 
-    if commit_file != "" and commit is not None:
-        bkp_commit_file = os.path.join(
-            os.path.dirname(commit_file),
-            f".{os.path.basename(commit_file)}.rapids-build-backend.bak",
-        )
+    if commit_file is not None and commit is not None:
```

Comment:
```suggestion
    if commit is not None:
```

Since `commit_file`'s default is `""`, it being `None` at this point in the code would be a bug, wouldn't it?  If so, I think we should remove this `commit_file is not None` condition, to limit the risk of accidentally, silently skipping writing the commit file.

Reply:
ah interesting, got it!

Well yeah if it's not possible (https://github.com/toml-lang/toml/issues/30) then I think this condition should be removed.